src/Arcus.Testing.Tests.Integration/Integration/DataFactory/Fixture/TemporaryDataFactoryDataFlow.cs

```diff
@@ -370,4 +409,94 @@ public async ValueTask DisposeAsync()
+        }
+
+        internal void ApplyOptions(ref DelimitedTextDataset dataSet, string sourceDataSetName)
+        {
+            string folderPathExpression = $"@concat('{sourceDataSetName}/', ";
+            foreach (var sourceDataSetParametersKey in SourceDataSetParameterKeyValues.Keys)
+            {
+                folderPathExpression += $"dataset().{sourceDataSetParametersKey}, '/', ";
+            }
+            // Remove the last ", '/', " part only if we have parameters
+            // Else just remove the last ", " part
+            folderPathExpression = (SourceDataSetParameterKeyValues.Keys.Count > 0 ? folderPathExpression[..^7] : folderPathExpression[..^2]) + ")";
+
+            if (SourceDataSetParameterKeyValues.Any())
+            {
+                dataSet.DataLocation.FolderPath = DataFactoryElement<string>.FromExpression(folderPathExpression);
+            }
+        }
+
+        internal void AddDataSetParameters(ref DelimitedTextDataset dataSet, IDictionary<string, string> sourceDataSetParameterKeyValues)
+        {
+            foreach (var parameters in sourceDataSetParameterKeyValues)
+            {
+                dataSet.Parameters.Add(parameters.Key, new EntityParameterSpecification(EntityParameterType.String));
+            }
+        }
+    }
+
+    public class TempDataFlowSinkOptions
+    {
+        /// <summary>
+        /// Gets the unique key and values of the parameters of the sink DataSet of the temporary DataFlow in Azure DataFactory.
+        /// </summary>
+        public IDictionary<string, string> SinkDataSetParameterKeyValues { get; } = new Dictionary<string, string>();
+
+        public TempDataFlowSinkOptions AddFolderPathParameters(IDictionary<string, string> parameters)
+        {
+            foreach (var parameter in parameters)
+            {
+                SinkDataSetParameterKeyValues.Add(parameter.Key, parameter.Value);
+            }
+            return this;
+        }
+
+        internal void ApplyOptions(ref DelimitedTextDataset dataSet, string sinkDataSetName)
+        {
+            string folderPathExpression = $"@concat('{sinkDataSetName}/', ";
+            foreach (var sinkDataSetParametersKey in SinkDataSetParameterKeyValues.Keys)
+            {
+                folderPathExpression += $"dataset().{sinkDataSetParametersKey}, '/', ";
+            }
+            // Remove the last ", '/', " part only if we have parameters
+            // Else just remove the last ", " part
+            folderPathExpression = (SinkDataSetParameterKeyValues.Keys.Count > 0 ? folderPathExpression[..^7] : folderPathExpression[..^2]) + ")";
+
+            if (SinkDataSetParameterKeyValues.Any())
+            {
+                dataSet.DataLocation.FolderPath = DataFactoryElement<string>.FromExpression(folderPathExpression);
+            }
+        }
+
+        internal void AddDataSetParameters(ref DelimitedTextDataset dataSet, IDictionary<string, string> sinkDataSetParameterKeyValues)
```

Comment:
Couldn’t we merge the `ApplyOptions` and this method together? You only need the `dataSet`, the other parameter here is already available in the options.

Reply:
Fixed in [#b0c4faf](https://github.com/arcus-azure/arcus.testing/pull/228/commits/b0c4faf4dc9bc205cb6af8c232fc2a03096d79a4)

Reply:
Fixed in [#386cc4e](https://github.com/arcus-azure/arcus.testing/pull/228/commits/386cc4e38fdfe46323a1d791df04f6826f06ed2b)